### ISSUE TITLE
Check for an extra leading directory in the uploaded dataset archives

### ISSUE THREAD

**Issue by zhiltsov-max** [cvat-ai/cvat#3849]
### My actions before raising this issue
- [ ] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [ ] Searched [past issues](/issues)

### Current Behaviour

Users often upload an archive with an extra leading directory, which is created by an archiver when a common approach is used: select files -> make archive in the file explorer menu. There is a problem with this approach that it prevents us from locating a dataset in multiple cases. Currently, the problem is partially mitigated by making search for datasets in Datumaro recursive, but in some formats we don't just call "detect dataset", but we also do some extra actions, involving file search, and these searches are not aligned with the recursive search logic.

### Possible Solution

Maybe, we should check the extracted files to only contain a single directory, and make the following actions inside that directory.

### Steps to Reproduce (for bugs)

1.
1.
1.
1.

### Your Environment

- Git hash commit (`git log -1`):
- Docker version `docker version` (e.g. Docker 17.0.05):
- Are you using Docker Swarm or Kubernetes?
- Operating System and version (e.g. Linux, Windows, MacOS):
- Code example or link to GitHub repo or gist to reproduce problem:
- Other diagnostic information / logs:
    <details>
    <summary>Logs from `cvat` container</summary>
    </details>

### Next steps
You may [join our Gitter](https://gitter.im/opencv-cvat/public) channel for community support.

**Comment by zhiltsov-max** [comment]
@kirill-sizov, @IRDonch

**Comment by IRDonch** [comment]
Makes sense to me. I think it's a better approach than trying to recursively search the archive for a dataset.

**Comment by zhiltsov-max** [comment]
Related https://github.com/openvinotoolkit/datumaro/pull/623

**Comment by zhiltsov-max** [comment]
Related: https://github.com/cvat-ai/cvat/issues/8991
Fixed in https://github.com/cvat-ai/cvat/pull/9209
